packages/itwinui-react/src/core/NonIdealState/NonIdealState.tsx

```diff
@@ -54,20 +70,69 @@ type NonIdealStateProps = {
  * <NonIdealState svg={<Svg404 />} heading='Not found' />
  */
 export const NonIdealState = React.forwardRef((props, forwardedRef) => {
-  const { className, svg, heading, description, actions, ...rest } = props;
+  const {
+    className,
+    svg,
+    heading,
+    description,
+    actions,
+    illustrationProps,
+    titleProps,
+    descriptionProps,
+    actionsProps,
+    ...rest
+  } = props;
 
   return (
     <Box
       className={cx('iui-non-ideal-state', className)}
       ref={forwardedRef}
       {...rest}
     >
-      <Box className='iui-non-ideal-state-illustration'>{svg}</Box>
-      {heading && <Box className='iui-non-ideal-state-title'>{heading}</Box>}
+      <Box
+        {...illustrationProps}
+        className={cx(
+          'iui-non-ideal-state-illustration',
+          illustrationProps?.className,
+        )}
+        ref={forwardedRef}
+        {...rest}
```

Comment:
`forwardedRef` and `...rest` should only be applied on the very top level element (lines 89-90), so it needs to be removed from all other elements
```suggestion
```

Reply:
fixed